### PR TITLE
[stable/openebs]: update openebs charts to 2.4.0

### PR DESCRIPTION
**Pull request by akhilerm**
- update chart version
- update values.yaml
- update README

Signed-off-by: Akhil Mohan <akhil.mohan@mayadata.io>

#### Special notes for your reviewer:
The PR should be squashed and merged.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/openebs]`)

### CODE REVIEW

File: charts/openebs/templates/daemonset-ndm.yaml [hunk 1]
```diff
@@ -29,6 +29,10 @@ spec:
         name: openebs-ndm
         openebs.io/version: {{ .Values.release.version }}
     spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
```
Comment (akhilerm) [inline charts/openebs/templates/daemonset-ndm.yaml:35]: @sonasingh46 @prateekpandey14  Should we use `with` or `if`.
- We need to be consistent across all the repos.
- We need to be consistent in the templates. For some fields `with` and  for some `if` are used.

Comment (survivant) [reply charts/openebs/templates/daemonset-ndm.yaml:35]: my two cents.  You should stay with Helm defaults.  If you do helm create test.. you will see that in the deployment file

````
{{- with .Values.imagePullSecrets }}
      imagePullSecrets:
        {{- toYaml . | nindent 8 }}
      {{- end }}
...
{{- with .Values.nodeSelector }}
      nodeSelector:
        {{- toYaml . | nindent 8 }}
      {{- end }}
      {{- with .Values.affinity }}
      affinity:
        {{- toYaml . | nindent 8 }}
      {{- end }}
      {{- with .Values.tolerations }}
      tolerations:
        {{- toYaml . | nindent 8 }}
      {{- end }}
````

**Comment by survivant** [comment]
will it include cstor-operator too ?

**Comment by akhilerm** [comment]
> will it include cstor-operator too ?

@survivant  [this](https://github.com/openebs/cstor-operators/pull/180) is the PR for cstor-operator charts

File: charts/openebs/templates/daemonset-ndm.yaml (hunk 1 above)
Comment (akhilerm) [reply charts/openebs/templates/daemonset-ndm.yaml:35]: great. Then I will follow this pattern,

**Pull request merged**